src/create-subscription-orders.js

```diff
@@ -0,0 +1,243 @@
+import fetch from 'isomorphic-fetch'
+import pMap from 'p-map'
+import parser from 'cron-parser'
+import { getSubscriptionConfig } from './config.js'
+import { updateOrderWithRetry } from './utils/utils.js'
+import {
+  ACTIVE_STATE,
+  ERROR_STATE,
+  REMINDER_SENT_STATE,
+  SEND_REMINDER_STATE,
+} from './states-constants.js'
+
+let apiRoot
+let ctpClient
+let logger
+let stats
+
+async function createSubscriptionOrders({
+  apiRoot: _apiRoot,
+  ctpClient: _ctpClient,
+  logger: _logger,
+  stateKeyToIdMap,
+}) {
+  stats = {
+    processedTemplateOrders: 0,
+    subscriptionOrderAlreadyCreated: 0,
+    successUrlCalls: 0,
+    recoverableErrorUrlCalls: 0,
+    unrecoverableErrorUrlCalls: 0,
+  }
+  apiRoot = _apiRoot
+  ctpClient = _ctpClient
+  logger = _logger
+  const stateIds = [
+    stateKeyToIdMap.get(ACTIVE_STATE),
+    stateKeyToIdMap.get(SEND_REMINDER_STATE),
+    stateKeyToIdMap.get(REMINDER_SENT_STATE),
+  ]
+  const orderQuery =
+    await _buildQueryForTemplateOrdersThatNeedSubscriptionOrders(stateIds)
+
+  for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
+    await pMap(templateOrders, _processTemplateOrder, { concurrency: 3 })
+
+  return stats
+}
+
+async function _processTemplateOrder({
+  id,
+  version,
+  orderNumber,
+  custom: { customFieldsRaw },
+  state: { key: stateKey },
+}) {
+  const config = getSubscriptionConfig()
```

Comment:
this and header creations is not changing at all so you might do this once instead of every iteration, wdyt ?

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/10/commits/e29ccf365bf0a44afa408f7f89cb51fcd8c0e9de